milestones/data.py

```diff
@@ -0,0 +1,373 @@
+# pylint: disable=no-member
+"""
+Application data management/abstraction layer.  Responsible for:
+
+1) Accessing information from various resources:
+* Internal application state  (through local models in models.py)
+* External application state  (ORM bindings with other apps, yuck)
+* Remote data services (through service adapters in resources.py)
+
+2) Calculating derivative information from existing state:
+* Algorithms and data manipulations
+* Aggregations
+* Annotations
+* Alternative data representations
+
+Accepts and returns standard Python data structures (dicts, arrays of dicts)
+for easy consumption and manipulation by callers -- the queryset stops here!
+
+Note the terminology difference at this layer vs. API -- create/fetch/update/delete/
+
+When the time comes for remote resources, import the module like so:
+if getattr(settings, 'TEST_MODE', False) or os.getenv('TRAVIS_MODE', False):
+    import milestones.tests.mocks.resources as remote
+else:
+    import milestones.resources as remote
+"""
+from . import exceptions
+from . import models as internal
+from . import serializers
+
+
+# PRIVATE/INTERNAL METHODS
+def _get_milestone_relationship_type(relationship):
+    """
+    Retrieves milestone relationship type object from backend
+    """
+    try:
+        return internal.MilestoneRelationshipType.objects.get(
+            name=relationship,
+            active=True
+        )
+    except internal.MilestoneRelationshipType.DoesNotExist:
+        if relationship in ['requires', 'fulfills']:
```

Comment:
I originally had a follow-on migration file (0002_seed_relationship_types.py) but it ended up being removed during one of the initial rounds of development.  I think in the beginning the thought was that the client would be determining the milestone relationship types based on various context-specific use cases.  However the approach has changed and it makes more sense to reinstate the data migration, which I'll do right now.